### PR TITLE
Fixed save and load NARRE model 

### PR DESCRIPTION
**Pull request by lthoang**
### Description

The current version cannot save NARRE model when specify `save_dir` in Experiment.

This change is mandatory for NARRE to save its weights.

### Checklist:

- [ ] I have added tests.
- [ ] I have updated the documentation accordingly.
- [ ] I have updated `README.md` (if you are adding a new model).
- [ ] I have updated `examples/README.md` (if you are adding a new example).
- [ ] I have updated `datasets/README.md` (if you are adding a new dataset).

### CODE REVIEW

**Comment by tqtg** [comment]
Can we elaborate more on the root cause? Is it because of changes in TensorFlow? Also, do we need to check the load function and make sure it works properly?

**Comment by lthoang** [comment]
> Can we elaborate more on the root cause? Is it because of changes in TensorFlow? Also, do we need to check the load function and make sure it works properly?

Both tensorflow model and optimizer need to be saved separately (detached from the model). For load function, the optimizer needs the saved weights for enabling adaptive learning rate approaches such as Adam and RMSProp.

File: cornac/models/narre/narre.py [hunk 1]
```diff
@@ -118,88 +175,39 @@ def __init__(self, n_users, n_items, vocab, global_mean, n_factors=32, embedding
...
-        Xu = layers.Dense(self.n_factors, use_bias=True, name="Xu")(
-            layers.Dropout(rate=self.dropout_rate, name="user_Oi")(
-                tf.reduce_sum(layers.Multiply()([user_attention, user_review_h]), 1)
-            )
-        )
-        Yi = layers.Dense(self.n_factors, use_bias=True, name="Yi")(
-            layers.Dropout(rate=self.dropout_rate, name="item_Oi")(
-                tf.reduce_sum(layers.Multiply()([item_attention, item_review_h]), 1)
-            )
+        self.graph = Model(
```
Comment (tqtg) [inline cornac/models/narre/narre.py:178]: Same comment as for the HDRModel

**Review by tqtg — APPROVED** [review]
